test/mocks/MockLimitOrderTaker.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { Ownable } from "contracts/abstracts/Ownable.sol";
+import { IStrategy } from "contracts/interfaces/IStrategy.sol";
+import { IUniswapRouterV2 } from "contracts/interfaces/IUniswapRouterV2.sol";
+import { MockERC1271Wallet } from "./MockERC1271Wallet.sol";
+
+contract MockLimitOrderTaker is IStrategy, MockERC1271Wallet {
```

Comment:
nit: I think in the future we could use a simpler mock contract instead of integrating real protocol, since we already decided to mock it 😆

Reply:
Yeah actually it's more like integration test to be honest. I also added the mock strategy contract (shared with GS) so we can separate these two cases as different type of test in the future.